src/controllers/events.js

```diff
@@ -29,3 +30,15 @@ exports.getRelatedEvents = async (req, res, next) => {
     .populate('ngo', 'name');
   return res.status(200).json({ success: true, data: relatedEvents });
 };
+
+exports.deleteEvent = async (req, res, next) => {
+  const event = await Event.findOne({ _id: req.params.id });
+  if (!event) {
+    return next(new ErrorResponse('No event found', 404));
+  }
+  if (event.ngo.toString() !== req.user._id.toString()) {
+    return next(new ErrorResponse('It is not your event', 401));
```

Comment:
Let's give them a nicer message, like `You don't have permissions to perform this operation`